### PR TITLE
Added FakeItEasy Extension methods and Unit tests.

### PR DESCRIPTION
**Pull request by BartDM**
# PR Details

Added FakeItEasy (FakeItEasy is a .Net dynamic fake framework for creating all types of fake objects, mocks, stubs etc. - https://fakeiteasy.readthedocs.io/en/stable/)

## Description

- Added a new project MockQueryable.FakeItEasy where I added the FakeItEasy extension methods
- Added a Test class MyServiceFakeItEasyTests where I duplicated the tests that were used in the MyServiceMoqTests class and converted them to use FakeItEasy
- Added CultureInfo in all test classes to provide correct parsing of DateTime when using this library.

## Related Issue

No related issue

## How Has This Been Tested

I added Unit tests to cover the FakeItEasy extension methods

## Checklist

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/MockQueryable/MockQueryable.FakeItEasy/MockQueryable.FakeItEasy.csproj [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <PackageReference Include="FakeItEasy" Version="5.1.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\MockQueryable\MockQueryable.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="System.Interactive.Async">
+      <HintPath>..\..\..\..\..\..\.nuget\packages\system.interactive.async\3.2.0\lib\netstandard2.0\System.Interactive.Async.dll</HintPath>
```
Comment (romantitov) [inline src/MockQueryable/MockQueryable.FakeItEasy/MockQueryable.FakeItEasy.csproj]: Please remove direct link to the package

File: src/MockQueryable/MockQueryable.Sample/MyServiceFakeItEasyTests.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+        [TestCase("AnyFirstName", "AnyExistLastName", "01/20/2012", "Users with DateOfBirth more than limit")]
+        [TestCase("ExistFirstName", "AnyExistLastName", "02/20/2012", "User with FirstName already exist")]
+        [TestCase("AnyFirstName", "ExistLastName", "01/20/2012", "User already exist")]
+        public void CreateUserIfNotExist(string firstName, string lastName, DateTime dateOfBirth, string expectedError)
+        {
+            //arrange
+
+            var userRepository = A.Fake<IUserRepository>();
+            //var userRepository = new Mock<IUserRepository>();
```
Comment (romantitov) [inline src/MockQueryable/MockQueryable.Sample/MyServiceFakeItEasyTests.cs]: Please remove the comment

**Comment by romantitov** [comment]
Thanks for so useful the pull request. 
Looks like 23 tests was failed with error: 

> Mapper already initialized. You must call Initialize once per application domain/process. 

I will be glad to include your changes to next release when you fix the issue with mapper and few my comments

**Comment by BartDM** [comment]
I've updated to code regarding your remarks.
The failing tests was a result of Automapper being initialized twice. 

Should be resolved now

**Comment by romantitov** [comment]
Thanks. Will try to find time complete the PR and create a new release this weekend

**Pull request merged**